### PR TITLE
Nginx update to ver 1.13.3

### PR DESCRIPTION
**Pull request by gaydabura**

### CODE REVIEW

**Comment by gaydabura** [comment]
updated PR. please review it

File: Dockerfile [hunk 1]
```diff
@@ -20,6 +20,7 @@ RUN addgroup -S -g 82 www-data && \
 RUN apk add --update \
 
         # Base packages
+        bash \
```
Comment (andypost) [inline Dockerfile]: please remove bash, alpine has busybox with `ash`

File: Dockerfile [hunk 2]
```diff
@@ -112,9 +109,19 @@ COPY nginx.conf /etc/nginx/nginx.conf
...
+RUN if [ -z $NGINX_FASTCGI_BUFFER_SIZE ] ; then export NGINX_FASTCGI_BUFFER_SIZE="32k"; fi
+RUN if [ -z $NGINX_FASTCGI_READ_TIMEOUT ] ; then export NGINX_FASTCGI_READ_TIMEOUT="900"; fi
+RUN if [ -z $NGINX_DOCUMENT_ROOT ] ; then export NGINX_DOCUMENT_ROOT="/var/www/html/"; fi
 
-WORKDIR /var/www/html
-VOLUME /var/www/html
+RUN sed -i "s!{{ NGINX_DOCUMENT_ROOT }}!${NGINX_DOCUMENT_ROOT}!g" /etc/nginx/conf.d/drupal8.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_BUFFERS }}/${NGINX_FASTCGI_BUFFERS}/" /etc/nginx/nginx.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_BUFFER_SIZE }}/${NGINX_FASTCGI_BUFFER_SIZE}/" /etc/nginx/nginx.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_READ_TIMEOUT }}/${NGINX_FASTCGI_READ_TIMEOUT}/" /etc/nginx/nginx.conf
```
Comment (andypost) [inline Dockerfile]: all this RUNs should be oneliner to have less layers in image

File: Dockerfile [hunk 3]
```diff
@@ -112,9 +109,19 @@ COPY nginx.conf /etc/nginx/nginx.conf
...
+
+
+WORKDIR ${NGINX_DOCUMENT_ROOT}
+VOLUME ${NGINX_DOCUMENT_ROOT}
```
Comment (andypost) [inline Dockerfile]: I see no reason to make it configurable with ENV because it will be set in stone in image anyway

File: drupal8.conf [hunk 4]
```diff
@@ -2,48 +2,46 @@ server {
...
     fastcgi_keep_conn on;
     fastcgi_index index.php;
     fastcgi_param QUERY_STRING $query_string;
     fastcgi_param SCRIPT_NAME $fastcgi_script_name;
     fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
-
     location / {
-
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Proto $scheme;
```
Comment (andypost) [inline drupal8.conf]: why that removed?

for every project we use following things to pass
```
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_cache_bypass $http_upgrade;
```

File: Dockerfile [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-ENV NGX_VER 1.10.1
+ENV NGX_VER 1.13.3
 ENV NGX_UP_VER 0.9.1
-ENV NGX_LUA_VER 0.10.5
-ENV NGX_NDK_VER 0.3.0
 ENV NGX_NXS_VER 0.54
-ENV LUAJIT_LIB /usr/lib/
-ENV LUAJIT_INC /usr/include/luajit-2.0/
+ENV NGINX_DOCUMENT_ROOT /var/www/html/
+ENV NGINX_FASTCGI_BUFFERS "16 32k"
```
Comment (andypost) [inline Dockerfile:7]: are you sure that 16 buffers enough?

**Review by andypost — CHANGES_REQUESTED** [review]
- You make too many layers in image
- remove bash
- no reason to configure volume&workdir with env
- add back proxy options

File: Dockerfile (hunk 5 above)
Comment (gaydabura) [reply Dockerfile:7]: by default `8 4k` and its value you can change locally for your project

File: drupal8.conf [hunk 6]
```diff
@@ -235,4 +212,5 @@ server {
         upload_progress_json_output;
         report_uploads uploads;
     }
-}
+
+}
```
Comment (andypost) [inline drupal8.conf:216]: (

File: Dockerfile [hunk 7]
```diff
@@ -112,9 +108,9 @@ COPY nginx.conf /etc/nginx/nginx.conf
 COPY fastcgi_params /etc/nginx/fastcgi_params
 COPY drupal* /etc/nginx/conf.d/
 
-
-WORKDIR /var/www/html
-VOLUME /var/www/html
+RUN if [ -z $NGINX_FASTCGI_BUFFERS ] ; then export NGINX_FASTCGI_BUFFERS="16 32k"; fi; if [ -z $NGINX_FASTCGI_BUFFER_SIZE ] ; then export NGINX_FASTCGI_BUFFER_SIZE="32k"; fi; if [ -z $NGINX_FASTCGI_READ_TIMEOUT ] ; then export NGINX_FASTCGI_READ_TIMEOUT="900"; fi; if [ -z $NGINX_DOCUMENT_ROOT ] ; then export NGINX_DOCUMENT_ROOT="/var/www/html/"; fi
+RUN sed -i "s!{{ NGINX_DOCUMENT_ROOT }}!${NGINX_DOCUMENT_ROOT}!g" /etc/nginx/conf.d/drupal8.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_BUFFERS }}/${NGINX_FASTCGI_BUFFERS}/;s/{{ NGINX_FASTCGI_BUFFER_SIZE }}/${NGINX_FASTCGI_BUFFER_SIZE}/;s/{{ NGINX_FASTCGI_READ_TIMEOUT }}/${NGINX_FASTCGI_READ_TIMEOUT}/" /etc/nginx/nginx.conf
```
Comment (andypost) [inline Dockerfile:113]: still lots of layers(

File: README.md [hunk 8]
```diff
@@ -1 +1,10 @@
-# skilld-docker-nginx
\ No newline at end of file
+# skilld-docker-nginx
+
+## Environment Variables
```
Comment (andypost) [inline README.md:3]: actually it is not true, because this vars applied only once at container build time

**Review by andypost — CHANGES_REQUESTED** [review]
Still require to minimize number of layers inside image

**Comment by alexkras** [comment]
Need to review last changes

File: drupal8.conf [hunk 9]
```diff
@@ -2,48 +2,56 @@ server {
...
     fastcgi_keep_conn on;
     fastcgi_index index.php;
     fastcgi_param QUERY_STRING $query_string;
     fastcgi_param SCRIPT_NAME $fastcgi_script_name;
     fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
-
     location / {
 
+        proxy_http_version 1.1;
+        proxy_set_header Upgrade $http_upgrade;
```
Comment (uzlov) [inline drupal8.conf:17]: lines 17-23 for proxying of websockets

**Comment by andypost** [comment]
I think it outdated enough

**Pull request closed without merging**